### PR TITLE
DB-Manager: strip comments from SQL

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #21271

### CODE REVIEW

File: python/plugins/db_manager/dlg_sql_window.py [hunk 1]
```diff
@@ -595,8 +595,14 @@ def createView(self):
     def _getSqlQuery(self):
```
Comment (Gustry) [inline python/plugins/db_manager/dlg_sql_window.py]: I think this function is used to store queries in preset or in file. They are many usages. So we will loose comments in saved file.

Comment (elpaso) [reply python/plugins/db_manager/dlg_sql_window.py]: @Gustry ok, I'll move it down into the executor

Comment (elpaso) [reply python/plugins/db_manager/dlg_sql_window.py]: @Gustry, looking better I thing you are wrong, the original behavior was altering the query by replacing newlines with spaces, and that was the source of the issue: https://github.com/qgis/QGIS/pull/9180/files#diff-9020444f364248acb9ad107f6fb38ba0L598

Comment (Gustry) [reply python/plugins/db_manager/dlg_sql_window.py]: `getSqlQuery` is used when we store the query as file: https://github.com/qgis/QGIS/blob/697237227d242c674940961b3374e9e55d928a0a/python/plugins/db_manager/dlg_sql_window.py#L234 so it means that the content of the file is not the same? (especially with this PR)

Sorry if I didn't understand. I will look later.

**Comment by elpaso** [comment]
@Gustry can I merge?

**Comment by Gustry** [comment]
As I said, the `_getSqlQuery` is called also when you save the query as a file or as a preset, so your PR removes comments from saved query.

In DB Manager:
```sql
-- this is a comment
SELECT * FROM foo;
```

Save this query as a preset or as a file, it becomes:
```sql
SELECT * FROM foo;
```

IMHO, we need another function like `getExecutableSqlQuery` which call first `getSqlQuery` and then clean the query (remove comments, replace `\n`, strip).

I noticed one regression from QGIS 2, comments on the same line are not working.
```sql
SELECT *  --all fields
FROM foo;
```
is not working in QGIS 3.4 while it's working fine in QGIS 2.18 and PgAdmin3 and 4.

**Comment by elpaso** [comment]
Superceeded by: https://github.com/qgis/QGIS/pull/9183

**Pull request closed without merging**